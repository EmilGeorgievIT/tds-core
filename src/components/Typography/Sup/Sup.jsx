import React from 'react'
import PropTypes from 'prop-types'

import safeRest from '../../../safeRest'

import textStyles from '../Text/Text.modules.scss'

const Sup = ({ children, size, ...rest }) => {
  return (
    <sup {...safeRest(rest)} className={textStyles[`${size}Sup`]} >
      {children}
    </sup>
  )
}


Sup.propTypes = {
  children: PropTypes.string.isRequired,
  size: PropTypes.oneOf([
    'small',
    'medium',
    'large'
  ])
}

Sup.defaultProps = {
  size: 'medium'
}

export default Sup
