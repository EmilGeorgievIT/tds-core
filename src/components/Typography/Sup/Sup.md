```
<div>
  <Paragraph><Text sup={String.fromCharCode(8482)}>Lorem</Text></Paragraph>
  <Paragraph size="small"><Text size="small" sup="Ipsum">Lorem</Text> dolor sit amet.</Paragraph>
  <Paragraph size="medium"><Text size="medium" sup="Ipsum">Lorem</Text> dolor sit amet.</Paragraph>
  <Paragraph size="large"><Text size="large" sup={String.fromCharCode(169)}>Lorem</Text> dolor sit amet.</Paragraph>
</div>
```
