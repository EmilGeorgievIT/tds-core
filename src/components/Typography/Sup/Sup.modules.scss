.base {
  font-size: 0.28rem;
}

.sup, .sub {
  font-size: .875rem;
  color: orange;
}

.sup {
  top: -.5em;
}

.sub {
  bottom: -.5em;
}
