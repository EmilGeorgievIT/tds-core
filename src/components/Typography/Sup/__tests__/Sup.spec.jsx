import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import Sup from '../Sup'

describe('Sup', () => {
  const doShallow = props => (
    shallow(<Sup {...props}>Some content</Sup>)
  )

  it('renders', () => {
    const sup = doShallow()

    expect(toJson(sup)).toMatchSnapshot()
  })

  it('renders an HTML sup tag', () => {
    const sup = doShallow()

    expect(sup).toHaveTagName('sup')
  })

  it('passes additional attributes to the sup element', () => {
    const sup = doShallow({ id: 'the-id' })

    expect(sup).toHaveProp('id', 'the-id')
  })

  it('does not allow custom CSS', () => {
    const sup = doShallow({ className: 'my-custom-class', style: { color: 'hotpink' } })

    expect(sup).not.toHaveProp('className', 'my-custom-class')
    expect(sup).not.toHaveProp('style')
  })
})
