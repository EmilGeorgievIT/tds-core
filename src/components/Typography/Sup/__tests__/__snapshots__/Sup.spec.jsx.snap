// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sup renders 1`] = `
<sup
  className="mediumSup"
>
  Some content
</sup>
`;
